packages/components/src/VirtualizedList/ListGrid/ListGrid.component.js

```diff
@@ -34,6 +35,14 @@ function ListGrid(props) {
 			getRowData: ({ index }) => collection[index],
 		});
 	}
+	let onRowClickCallback;
+	if (onRowClick) {
+		onRowClickCallback = decorateRowClick(onRowClick);
+	}
+	let onRowDoubleClickCallback;
+	if (onRowDoubleClick) {
+		onRowDoubleClickCallback = decorateRowDoubleClick(onRowDoubleClick);
+	}
```

Comment:
Can't you do those checks directly on `decorateRowClick` and `decorateRowDoubleClick` and call the function directly?